#!/usr/bin/env python
PACKAGE="color_calibration"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

## Thresholding parameters
gen.add("r_min", int_t, 0, "Lower Bound of R", 0, 0, 255);
gen.add("r_max", int_t, 0, "Higher Bound of R", 0, 0, 255);
gen.add("g_min", int_t, 0, "Lower Bound of G", 0, 0, 255);
gen.add("g_max", int_t, 0, "Higher Bound of G", 0, 0, 255);
gen.add("b_min", int_t, 0, "Lower Bound of B", 0, 0, 255);
gen.add("b_max", int_t, 0, "Higher Bound of B", 0, 0, 255);

## Morphological parameters
gen.add("opening_mat_point", int_t, 0, "Center of the matrix for the opening operation (size extrapolated 2x+1)", 1, 1, 7);
gen.add("opening_iter", int_t, 0, "Iterations of opening applied on the thresholded image", 3, 0, 10);
gen.add("closing_mat_point", int_t, 0, "Center of the matrix for the closing operation (size extrapolated 2x+1)", 1, 1, 7);
gen.add("closing_iter", int_t, 0, "Iterations of closing applied on the opened thresholded image", 0, 0, 10);

## Bilateral filter
gen.add("bilateral_iter", int_t, 0, "Iterations of bilateral applied on the enhanced image", 0, 0, 10)

## Dump Parameters
gen.add("save_params", bool_t, 0, "Bool param to save the parameters")

exit(gen.generate(PACKAGE, "color_calibrate", "vision"));
